### ISSUE TITLE
Switch KaTeX to MathJax

### ISSUE THREAD

**Issue by Suyun514** [marktext/marktext#1876]
### Describe your feature request

Mark Text use KaTeX to display math expressions, but KaTeX is too weak to display some complex expressions, such as `\ce`。

### What problem does this feature solve? [optional]

Displaying some complex expressions.

**Comment by RyanGreenup** [comment]
Most things can be done with KaTeX, according to the [docs](https://katex.org/docs/support_table.html#c) carbon bonds are supported. 

The only big missing feature is equation numbering, but that’s tricky to do in markdown because the ` align ` environment shouldn’t be used inside ` $$ ` or ` \[ ` and last I checked ` pandoc ` doesn’t inteligently remove ` $$ ` where they shouldn’t be used. So if MathJax is used special attention would have to be paid to this like in RMarkDown or ` org-mode ` where ` \begin{align} ` May de limit a math environment . In order to achieve numbering and alignement in KaTeX there’s the ` aligned ` environment which doesn’t support tags or labels in LaTeX ( but confusingly supports `  \tags{} `  in KaTeX) which can be used with something like ` \ldots (1) ` in order to achieve compatibility with LaTeX through `  pandoc ` 

Another thing to consider is speed and portability, Either mathjax would have to be installed locally or a consistent online connection would be required to access the CDN and even then from what I understand it does render slower.

such a switch would need to be done in a way that doesn’t slow everything down and doesn’t break pdf export.

**Comment by pelegm** [comment]
There are numerous bugs in katex compilations of simple equations, and I am wondering whether it is katex's fault of marktext's fault. Here's a quick example:

![Screenshot from 2020-03-01 16-12-37](https://user-images.githubusercontent.com/3422626/75627318-b8845d00-5bd7-11ea-8076-b4e15f575346.png)

There's a missing square root sign there. Going in and out from the equation, solves the issue temporarily:

![Screenshot from 2020-03-01 16-12-51](https://user-images.githubusercontent.com/3422626/75627330-ce921d80-5bd7-11ea-81b8-4c8331732315.png)

But then switching tabs and coming back the square root disappears again.

This happens with numerous math expressions in my notes, and does not happen in Typora, which as far as I know uses mathjax. I really don't know if this is a katex issue, however.

**Comment by pelegm** [comment]
KaTeX, or at least the version used by marktext, does not seem to support `\substack`, which is a shame.

**Comment by blsz** [comment]
Please keep KaTeX.  The output looks better than MathJax and it is faster.

**Comment by fxha** [comment]
We keep KaTeX but the we noticed the display issue too. The issue is tracked in #1951.
